lib/capistrano/asg/tasks/rolling.rake

```diff
@@ -106,11 +106,16 @@ namespace :rolling do
       end
     end
 
-    instances = config.instances.auto_terminate
-    if instances.any?
-      logger.info 'Terminating instance(s)...'
-      instances.terminate
+    begin
+      instances = config.instances.auto_terminate
+      if instances.any?
+        logger.info 'Terminating instance(s)...'
+        instances.terminate
+      end
+    rescue Aws::Errors::ServiceError => e
+      logger.warning "Error deleting instance: #{e}"
     end
+    invoke 'rolling:instance_refresh_status' if fetch(:asg_wait_for_instance_refresh)
```

Comment:
You are invoking another rake task from the `rolling:cleanup` task, which works. But, the cleanup task is also called when the deployment fails (`deploy:failed`). In this case we don't want to monitor the instance refresh, because it hasn't started.

Not sure what the best way would be to fix this, but perhaps you could check if there were any instance refreshes started before invoke.

Reply:
Updated and tested locally :+1: